index.d.ts

```diff
@@ -65,17 +63,27 @@ export function consumeAllItems() : Promise<void>;
 
 /**
  * Get a list of products (consumable and non-consumable items, but not subscriptions)
- * @param {string[]} skus The item skus
- * @returns {Promise<Product[]>}
+ * @param skus The item skus
  */
-export function getProducts(skus: string[]) : Promise<Product[]>;
+export function getProducts<A extends string, B extends string, C extends string, D extends string, E extends string, F extends string>(skus: [A, B, C, D, E, F]): Promise<[Product<A>, Product<B>, Product<C>, Product<D>, Product<E>, Product<F>]>;
+export function getProducts<A extends string, B extends string, C extends string, D extends string, E extends string>(skus: [A, B, C, D, E]): Promise<[Product<A>, Product<B>, Product<C>, Product<D>, Product<E>]>;
+export function getProducts<A extends string, B extends string, C extends string, D extends string>(skus: [A, B, C, D]): Promise<[Product<A>, Product<B>, Product<C>, Product<D>]>;
+export function getProducts<A extends string, B extends string, C extends string>(skus: [A, B, C]): Promise<[Product<A>, Product<B>, Product<C>]>;
+export function getProducts<A extends string, B extends string>(skus: [A, B]): Promise<[Product<A>, Product<B>]>;
+export function getProducts<A extends string>(skus: [A]): Promise<[Product<A>]>;
+export function getProducts(skus: string[]): Promise<Product<string>[]>;
```

Comment:
It makes `getProducts(['a', 'b'])` return a tuple of `Product<'a'>` and `Product<'b'>`, instead of an array of generic products.

That way the following will be correctly caught as errors:

```ts
const products = await getProducts(['a', 'b'])

products[2] // TS will complain since it's only a 2 element tuple

if (products[0].productId === 'c') // TS will complain since none of the productIds can be 'c'
```